nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/PutAzureEventHub.java

```diff
@@ -131,18 +132,20 @@
         return propertyDescriptors;
     }
 
+    private ScheduledExecutorService executor;
+
     @OnScheduled
     public final void setupClient(final ProcessContext context) throws ProcessException{
         final String policyName = context.getProperty(ACCESS_POLICY).getValue();
         final String policyKey = context.getProperty(POLICY_PRIMARY_KEY).getValue();
         final String namespace = context.getProperty(NAMESPACE).getValue();
         final String eventHubName = context.getProperty(EVENT_HUB_NAME).getValue();
 
-
         final int numThreads = context.getMaxConcurrentTasks();
         senderQueue = new LinkedBlockingQueue<>(numThreads);
+        executor = Executors.newScheduledThreadPool(numThreads);
         for (int i = 0; i < numThreads; i++) {
-            final EventHubClient client = createEventHubClient(namespace, eventHubName, policyName, policyKey);
+            final EventHubClient client = createEventHubClient(namespace, eventHubName, policyName, policyKey, executor);
```

Comment:
the use of this executor here too gives me pause.  I'm not sure we're using the nifi api correctly here if we're doing this.